src/Admin/fields/AssociatedDonor/index.tsx

```diff
@@ -0,0 +1,52 @@
+import AdminSection, {AdminSectionField} from '@givewp/components/AdminDetailsPage/AdminSection';
+import {__} from '@wordpress/i18n';
+import {useFormContext, useFormState} from 'react-hook-form';
+import {AsyncPaginate} from 'react-select-async-paginate';
+import styles from './styles.module.scss';
+import {DonorOption} from './types';
+import {useDonorAsyncSelect} from './useDonorAsyncSelect';
+
+/**
+ * @unreleased add mode prop
+ * @since 4.9.0 Add error prop to all AdminSectionField components
+ * @since 4.8.0 updated to async donor dropdown
+ * @since 4.6.0
+ */
+export default function AssociatedDonor({mode}: {mode: 'test' | 'live'}) {
+    const {watch, setValue} = useFormContext();
+    const {errors} = useFormState();
+    const donationDonorId = watch('donorId');
+
+    const {selectedOption, loadOptions, mapOptionsForMenu, error} = useDonorAsyncSelect(donationDonorId || null, mode);
+
+    const handleDonorChange = (selectedOption: DonorOption | null) => {
+        setValue('donorId', selectedOption?.value ?? null, {shouldDirty: true});
+    };
+
+    return (
+        <AdminSectionField error={errors.donorId?.message as string}>
+            <label htmlFor="donorId">{__('Donor', 'give')}</label>
+            <p className={styles.fieldDescription}>{__('Link the donation to the selected donor', 'give')}</p>
```

Comment:
This description will be different between screens so we should add a prop for `description`.  

Also, where is the `styles.fieldDescription` coming from?

Reply:
Resolved: https://github.com/impress-org/givewp/pull/8119/commits/ef2bce1a9cb856c9258fe1bf2a2f9403765a9435

Reply:
Thanks! Resolved 5e491649